src/Bonnier/BCM/Services/MetaTag.php

```diff
@@ -0,0 +1,198 @@
+			}
+			
+			// add tablet breaking point
+			if (self::$objSettings->tablet_breakpoint) {
+				$arrTags['bcm-tablet-breakpoint'] = self::$objSettings->tablet_breakpoint;
+			}
+			
+			// add mobile breaking point
+			if (self::$objSettings->mobile_breakpoint) {
+				$arrTags['bcm-mobile-breakpoint'] = self::$objSettings->mobile_breakpoint;
+			}
+			
+			// we only apply article tags when necessary
+			if (!is_front_page() && (is_singular() || is_single())) {
+				$arrTags = array_merge($arrTags, self::get_article_tags());
+			}
+			
+			// write the actual tags
+			self::write_meta_tags($arrTags);
+		}
+	}
+	
+	/**
+	 * Get specific article meta tags
+	 *
+	 * @return array
+	 */
+	private static function get_article_tags() {
+		$arrArticleTags = [
+			'bcm-title' => self::get_bcm_title()
+		];
+
+		// set content type
+		if ($strContentType = self::get_bcm_content_type()) {
+			$arrArticleTags['bcm-content-type'] = $strContentType;
+		}
+		
+		// set all categories
+		if ($arrCategories = self::get_bcm_categories()) {
+			$arrArticleTags['bcm-categories'] = implode(',', $arrCategories);
+		}
+		
+		// set all tags
+		if ($arrTags = self::get_bcm_tags()) {
+			$arrArticleTags['bcm-tags'] = implode(',', $arrTags);
+		}
+		
+		// set advertorial type
+		if ($strAdvertorialType = self::get_bcm_advertorial_type()) {
+			$arrArticleTags['bcm-advertorial-type'] = implode(',', $strAdvertorialType);
+		}
+		
+		// set advertorial label
+		if ($strAdvertorialLabel = self::get_bcm_advertorial_label()) {
+			$arrArticleTags['bcm-advertorial-label'] = implode(',', $strAdvertorialLabel);
+		}
+		
+		return $arrArticleTags;
+	}
+	
+	/**
+	 * Get title from post
+	 *
+	 * @return string
+	 */
+	private static function get_bcm_title() {
+		global $post;
+		return apply_filters('wp_bcm_set_title', $post->post_title);
+	}
+	
+	/**
+	 * Get content type
+	 *
+	 * @return string
+	 */
+	private static function get_bcm_content_type() {
+		$strContentType = null;
+		return apply_filters('wp_bcm_set_content_type', $strContentType);
+	}
+	
+	/**
+	 * Get advertorial content type
+	 *
+	 * @return string
+	 */
+	private static function get_bcm_advertorial_type() {
+		$strAdvertorialContentType = null;
+		return apply_filters('wp_bcm_set_advertorial_type', $strAdvertorialContentType);
+	}
+	
+	/**
+	 * Get advertorial content label
+	 *
+	 * @return string
+	 */
+	private static function get_bcm_advertorial_label() {
+		$strAdvertorialContentLabel = null;
+		return apply_filters('wp_bcm_set_advertorial_label', $strAdvertorialContentLabel);
+	}
+	
+	/**
+	 * Get array of categories name from post
+	 *
+	 * @return array
+	 */
+	private static function get_bcm_categories() {
+		global $post;
+		
+		$arrCategories = [];
+		$arrPostCategories = wp_get_post_terms($post->ID, 'category');
+
+		foreach ($arrPostCategories as $arrCategory) {
+			$arrCategories[] = $arrCategory->name;
+		}
+		
+		return apply_filters('wp_bcm_set_categories', $arrCategories);
```

Comment:
Just a suggestion a more functional approach that would make the code more expressive and save you a lot of un needed variables: 
``` php
return apply_filters('wp_bcm_set_categories', array_map(funnction($category){
    return $category->name;
}, wp_get_post_terms(get_post()->ID, 'category')));
```